server/src/com/cloud/network/guru/GuestNetworkGuru.java

```diff
@@ -396,6 +396,16 @@ public NicProfile allocate(final Network network, NicProfile nic, final VirtualM
         return nic;
     }
 
+    /**
+     * Return a string representing network Cidr for the specifeid network
+     * @param guestNetwork
+     * @return valid network Cidr for the specified network
+     */
+    protected String getValidNetworkCidr(Network guestNetwork){
```

Comment:
I think you should extract this method to another common class (e.g. NetworkModel) since this method can be reused over the different classes.